test/shared/src/main/scala/zio/test/TestRuntime.scala

```diff
@@ -0,0 +1,108 @@
+package zio.test
+
+import zio.Exit
+import zio.ZIO
+import zio.internal.Executor
+import zio.random.Random
+import zio.stream.ZStream
+
+object TestRuntime {
+
+  /**
+   * Returns multiple possible outcomes of the given `zio`.
+   */
+  def paths[R <: Random, E, A](zio: ZIO[R, E, A]): ZStream[R, Nothing, Exit[E, A]] = {
+
+    val yielding = yieldingEffects(zio)
+
+    val singleRun = runOnce(yielding)
+
+    ZStream.repeatEffect(singleRun)
+  }
+
+  private def runOnce[R <: Random, E, A](zio: ZIO[R, E, A]) =
+    makeTestExecutor.flatMap(yieldingEffects(zio).lock(_).run)
+
+  private val makeTestExecutor: ZIO[Random, Nothing, Executor] =
+    ZIO.effectTotal {
+      new Executor {
+        val pendingRunnables = new java.util.Vector[Runnable]
+        var isRunning = false
+
+        def here         = true
+        def metrics      = None
+        def yieldOpCount = Int.MaxValue
+        def submit(runnable: Runnable): Boolean = synchronized {
+          if (isRunning)
+            pendingRunnables.add(runnable)
+          else {
+            isRunning = true
+            runnable.run()
+
+            while (! pendingRunnables.isEmpty()) {
+              val randomIndex = util.Random.nextInt(pendingRunnables.size())
+              pendingRunnables.remove(randomIndex).run()
+            }
+            
+            isRunning = false
+          }
+          true
+        }
+    }
+  }
+
+  private def yieldingEffects[R, E, A](zio: ZIO[R, E, A]): ZIO[R, E, A] =
```

Comment:
What else do we need to do here?

Reply:
I think everything is done here. Would be good to rise the confidence.